core/shared/src/main/scala/zio/config/WriteModule.scala

```diff
@@ -69,11 +69,9 @@ private[config] trait WriteModule extends ConfigDescriptorModule {
           val rightResult = go(cd.right, tuple._2)
 
           for {
-            left   <- leftResult
-            right  <- rightResult
-            merged  = left.merge(right)
-            result <- merged.headOption.toRight("Failed to write the config back to property tree, at zip node")
-          } yield result
+            left  <- leftResult
+            right <- rightResult
```

Comment:
I can’t remember the top of my head while I was writing this loop. But just by looking at the change, I think it makes sense.